src/levanter/layers/attention.py

```diff
@@ -1183,8 +1196,10 @@ def __call__(self, x: NamedArray, mask: Optional[NamedArray | AttentionMask], *,
 
         # Apply rotary position embeddings if configured
         if self.config.rope is not None:
-            rot_embs = self.config.rope.build(self.config.HeadSize, q.resolve_axis("position"))
-            q, k = rot_embs(self.config.HeadSize, q, k)
+            pos_ids = pos_ids or hax.arange(x.resolve_axis("position"))
+            rot_embs = self.config.rope.build(self.config.HeadSize)
```

Comment:
Since all the args for this are now available at init time, it feels like this should be built inside `init` rather than inside `__call__`